kale/loaddata/multi_domain.py

```diff
@@ -229,3 +232,249 @@ def _split_dataset_few_shot(dataset, n_fewshot, random_state=None):
+    domains = np.unique(domain_labels)
+    subset_idx = [[] for i in range(n_partitions)]
+    for domain_label_ in domains:
+        domain_idx = np.where(domain_labels == domain_label_)[0]
+        subsets = split_by_ratios(torch.from_numpy(domain_idx), split_ratios)
+        for i in range(n_partitions):
+            subset_idx[i].append(domain_idx[subsets[i].indices])
+
+    for i in range(n_partitions):
+        subset_idx[i] = np.concatenate(subset_idx[i])
+
+    return subset_idx
+
+
+class MultiDomainImageFolder(VisionDataset):
+    """A generic data loader where the samples are arranged in this way: ::
+            root/domain_a/class_x/xxx.ext
+            root/domain_a/class_x/xxy.ext
+            ...
+            root/domain_a/class_y/xxz.ext
+            ...
+            root/domain_k/class_x/123.ext
+            root/domain_k/class_x/abc3.ext
+            ...
+            root/domain_k/class_y/asd932_.ext
+        Args:
+            root (string): Root directory path.
+            loader (callable): A function to load a sample given its path.
+            extensions (tuple[string]): A list of allowed extensions.
+                both extensions and is_valid_file should not be passed.
+            transform (callable, optional): A function/transform that takes in a sample and returns a transformed
+                version.  E.g, ``transforms.RandomCrop`` for images.
+            target_transform (callable, optional): A function/transform that takes in the target and transforms it.
+            sub_domain_set (list): A list of domain names, which should be a subset of domains in image folders.
+                Defaults to None.
+            sub_class_set (list): A list of class names, which should be a subset of classes in image folders.
+                Defaults to None.
+            is_valid_file (callable, optional): A function that takes path of a file and check if the file is a valid
+                file (used to check of corrupt files) both extensions and is_valid_file should not be passed.
+         Attributes:
+            classes (list): List of the class names sorted alphabetically.
+            class_to_idx (dict): Dict with items (class_name, class_index).
+            samples (list): List of (sample path, class_index) tuples
+            targets (list): The class_index value for each image in the dataset
+            domains (list): List of the domain names sorted alphabetically.
+            domain_to_idx (dict): Dict with items (domain_name, domain_index).
+            domain_labels (list): The domain_index value for each image in the dataset
+        """
+
+    def __init__(
+        self,
+        root: str,
+        loader: Callable[[str], Any] = default_loader,
+        extensions: Optional[Tuple[str, ...]] = IMG_EXTENSIONS,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        sub_domain_set=None,
+        sub_class_set=None,
```

Comment:
white space around '=' for the two lines above.

Reply:
It is modified by pre-commit